### PR TITLE
Fix back button functionality

### PR DESCRIPTION
**Pull request by chowington**
Fixes https://github.com/VEuPathDB/web-eda/issues/590

### CODE REVIEW

File: src/lib/core/components/visualizations/VisualizationsContainer.tsx [hunk 1]
```diff
@@ -32,6 +39,18 @@ import { GeoConfig } from '../../types/geoConfig';
 
 const cx = makeClassNameHelper('VisualizationsContainer');
 
+// Doesn't seem to work
```
Comment (bobular) [inline src/lib/core/components/visualizations/VisualizationsContainer.tsx:42]: Do you mean the TS typing doesn't work?

Comment (chowington) [reply src/lib/core/components/visualizations/VisualizationsContainer.tsx:42]: The typing works fine. This component simply doesn't fix the problem I wanted it to!

**Comment by chowington** [comment]
Do test this "in anger" (@bobular ;)), as in my own earlier testing, the `locationKeys` and `pathnamesRef` arrays would sometimes get into weird states, opening up the potential for unexpected history navigation behavior. That's why I added the sanity checks to the code. My latest testing seemed to work well though.

**Review by bobular — APPROVED** [review]
Extensively tested, including the "back into new" scenario.

Looks nice - can't find any problems :-)

**Comment by dmfalke** [comment]
Apologies, I started testing the before I went on vacation and found a small bug. I thought I made a comment about it, but I don't see it. I'll provide more details soon.

**Review by dmfalke — APPROVED** [review]
I'm not able to reproduce the issue I remember hitting (that's what I get for not reporting it immediately :slightly_smiling_face:), so I think this is good to go! Nice work!

File: src/lib/workspace/WorkspaceRouter.tsx [hunk 2]
```diff
@@ -62,6 +63,76 @@ export function WorkspaceRouter({
...
+
+          // Sanity check
+          if (!(pathnamesCursorRef.current > 0)) return;
+
+          pathnamesCursorRef.current--;
+
+          const lastPathname =
+            pathnamesRef.current[pathnamesCursorRef.current + 1];
+          const newAnalysisRegex = /eda\/.*\/new\/.*/;
+          const savedAnalysisRegex = /eda\/[^/]*\/(?!new\/)([^/]*)\/.*/;
```
Comment (jtlong3rd) [inline src/lib/workspace/WorkspaceRouter.tsx:114]: The base url for the EDA workspace isn't necessarily `/eda` (for example, the ClinEpi website uses `/workspace/analyses`), so these regexes will need to be dynamic based on the base url.

**Review by jtlong3rd — COMMENTED** [review]
Just one inline suggestion, and then this is likely good to merge!

File: src/lib/workspace/WorkspaceRouter.tsx (hunk 2 above)
Comment (dmfalke) [reply src/lib/workspace/WorkspaceRouter.tsx:114]: Good catch! :detective:

Comment (chowington) [reply src/lib/workspace/WorkspaceRouter.tsx]: @jtlong3rd Thanks for the info! How does it look now?

Comment (jtlong3rd) [reply src/lib/workspace/WorkspaceRouter.tsx]: The regexes look great! Just to be safe, could you also add `path` to the effect dependencies?

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**